### PR TITLE
Reintroduced pagination to the products page

### LINKED ISSUE #446: Revert pagination for products page

### PR DESCRIPTION
**Pull request by ValKoDim**
I reverted the changes to pagination to the products page. Also made some adjustments to our mock backend, to better replicate the backend logic.

https://github.com/user-attachments/assets/c68b5bf6-fcec-44b3-9a42-6ccbccf5faf7

### CODE REVIEW

File: code/src/pages/products/ProductsPage.tsx [hunk 1]
```diff
@@ -32,12 +34,13 @@ const ProductsPage = () => {
...
     tags: categoryType ? `category:${categoryType}` : "",
     page: page - 1,
-    size: 1000, //The number 1000 is temporary solution, until we implement endless scrolling.
+    size: 10,
     sort: sortOption ?? "recommended",
     lang: locale
   });
 
   const productsList = productsResponse?.content;
+  console.log(productsList);
```
Comment (heorhiiss) [inline code/src/pages/products/ProductsPage.tsx]: Remove `console.log(productsList)`

**Review by mhevyk — APPROVED** [review]

**Comment by OliaHrytsak** [comment]
This is how the last page looks like
Its not good. Try to fix it
image

**Review by OliaHrytsak — CHANGES_REQUESTED** [review]
This is how the last page looks like
Its not good. Try to fix it
image

File: code/src/pages/products/ProductsPage.tsx (hunk 1 above)
Comment (ValKoDim) [reply code/src/pages/products/ProductsPage.tsx]: Done

**Comment by ValKoDim** [comment]
> This is how the last page looks like Its not good. Try to fix it image

@OliaHrytsak   Check changes

**Review by OliaHrytsak — APPROVED** [review]

**Review by heshvab — APPROVED** [review]

**Pull request merged**